### ISSUE TITLE
TODOs before 0.15 release

### ISSUE THREAD

**Issue by NicolasHug** [pytorch/vision#7217]
This is a meta-issue to keep track of all the things we need to do before the branch cut / release. Core branch cut is planned for Monday ~13~ 15th.

This superseeds https://github.com/pytorch/vision/pull/7092 and includes stuff unrelated to transformsV2. @pmeier @vfdev-5 please feel free to directly edit this issue with updates, and add anything I may have missed

### Transforms V2 Stuff
- [x]  https://github.com/pytorch/vision/pull/7215 @pmeier 
- [x] Add tests for class transforms (type preservation, input / output format preservation (tuple, dict), pass-through non-native types). @pmeier #7238 
  - [x] https://github.com/pytorch/vision/pull/7276
- [x] https://github.com/pytorch/vision/pull/7211 @vfdev-5 
- [x]  https://github.com/pytorch/vision/pull/6663 @pmeier 
  - [x] Decide on name and location for dataset wrapper - https://github.com/pytorch/vision/pull/7279
- [x]  Address all TODOs in https://github.com/pytorch/vision/pull/7159 @pmeier @vfdev-5 @NicolasHug
  - [x] Fix `fill` parameter compatibilty with torchscript https://github.com/pytorch/vision/pull/7252  @pmeier 
  - [x] https://github.com/pytorch/vision/pull/7253
  - [x] https://github.com/pytorch/vision/pull/7254
  - [x] https://github.com/pytorch/vision/pull/7255
  - [x] https://github.com/pytorch/vision/pull/7256
  - [x] https://github.com/pytorch/vision/pull/7257
  - [x] https://github.com/pytorch/vision/pull/7258
  - [x] https://github.com/pytorch/vision/pull/7271
- [x] make sure there's no critical TODOs left in https://github.com/pytorch/vision/pull/7092 and https://github.com/pytorch/vision/pull/7082 @NicolasHug
- [x] Avoid removing BBoxes on transforms, just mark them as degenerate https://github.com/pytorch/vision/pull/7246 @NicolasHug
  - [x] https://github.com/pytorch/vision/pull/7268 
- [x] https://github.com/pytorch/vision/pull/7223  @NicolasHug 
- [x]  Confirm training references - https://github.com/pytorch/vision/pull/7220 @NicolasHug (Done, just haven't posted results yets)
- [x] make `Image(some_pil_image)` work (description: https://github.com/pytorch/vision/pull/6663#discussion_r1093121007), @vfdev-5 @pmeier, https://github.com/pytorch/vision/pull/7231
- [ ] Figure out a better name than `convert_dtype` https://github.com/pytorch/vision/pull/7092/files#diff-0c3dc07ddb80b9a83bf674efe1dcaee6aaabab59210d4916d2e35536c498710aR441
- [x] Actually migrate to `torchvision.transforms.v2` (or whatever else if we change our mind) https://github.com/pytorch/vision/issues/7097 @pmeier @vfdev-5 @NicolasHug #7261
  - [x] Double check what we expose as public
  - [x] https://github.com/pytorch/vision/pull/7278
- [x] https://github.com/pytorch/vision/pull/7154 @pmeier
- [x] Figure out why CI didn't trigger on wrong kernel call from datapoint: https://github.com/pytorch/vision/pull/7215#discussion_r1102818876 @pmeier #7226

### Other stuff

- [x]  https://github.com/pytorch/vision/pull/7160 @NicolasHug 
- [x] https://github.com/pytorch/vision/issues/7205 @pmeier 
  - [x] and add int support to V2  https://github.com/pytorch/vision/pull/7248 @pmeier
- [x] Fix https://github.com/pytorch/vision/issues/7103 (CC @oke-aditya  :) )
- [x] https://github.com/pytorch/vision/issues/7247 @pmeier @NicolasHug 

### If time allows

- [ ] Decide if we want to remove `None` as option to `antialias`
- [ ] Properly investigate and address https://github.com/pytorch/vision/issues/7143
- [ ] Now that https://github.com/pytorch/pytorch/pull/90771 is merged, let `Resize()` rely on `interpolate()`'s native `uint8` handling instead of converting to and from `float`. 
- [ ] https://github.com/pytorch/vision/pull/6960
- [ ] Bikeshed on whether `format` and `spatial_size` are adequate meta-data names for bboxes https://github.com/pytorch/vision/pull/7092/files#r1071959354
- [ ] Decide whether it's worth enforcing that only one `BoundingBox` instance is present per sample. Some transforms enforce that, some don't. It could simplify some checks - sort of related: https://github.com/pytorch/vision/pull/7092/files#r1071600864
- [x] Make the dataset wrapper work properly on subclasses of builtin datasets. @pmeier #7236
- [X] Make `datapoints.FillType` and the others private @pmeier https://github.com/pytorch/vision/pull/7266

## After branch cut, before release

- [X] List out all the current design decision that may change and / or that we actively need user feedback on @NicolasHug  https://github.com/pytorch/vision/issues/7319
- [x] Write docs for V2 transforms: 
  - [x] Docstrings
  - [x] Tutorial(s)
- [X] Write summary of design changes @pmeier @NicolasHug  https://github.com/pytorch/vision/issues/7384
- [X] Update previously released content (video, blog posts etc.) to reflect latest changes (typically the V2 are now JIT-scriptable, there's no Label anymore, etc.)
- [x] Write release notes @NicolasHug 
- [x] https://github.com/pytorch/vision/issues/7265 - https://github.com/pytorch/vision/pull/7330

## Just after release

- [ ] Port training references to V2? Would it require releasing new models?

**Comment by justinchuby** [comment]
Linking #7056 for visibility
